elasticdl/python/master/servicer.py

```diff
@@ -170,7 +174,8 @@ def _update_edl_embedding_table(self, name_var_list):
             )
 
     def _update_model(self):
-        assert self._lock.locked()
+        if not self._use_async and not self._lock.locked():
+            raise RuntimeError("In sync mode, lock is not acquired")
```

Comment:
How about "Lock must be acquired when updating the model in sync mode"?

Reply:
It seems better.